jspredict.js

```diff
@@ -137,8 +145,9 @@
       var transits = [];
       var nextTransit;
       var iterations = 0;
+      var transitsCounted = 0;
 
-      while (iterations < max_iterations) {
+      while (iterations < max_iterations && transitsCounted < maxTransits) {
```

Comment:
We could get rid of the `transitsCounted` var if we used `transits.length < maxTransits` here instead